internal/tls/option_test.go

```diff
@@ -325,3 +325,78 @@ func TestWithTLSConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestWithInsecureSkipVerify(t *testing.T) {
+	type T = credentials
+	type args struct {
+		insecure bool
+	}
+	type want struct {
+		obj *T
+		err error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, *T, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+
+	defaultCheckFunc := func(w want, obj *T, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if !reflect.DeepEqual(obj, w.obj) {
+			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", obj, w.obj)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "InsecureSkipVerify is true",
+			args: args{
+				insecure: true,
+			},
+			want: want{
+				obj: &T{
+					insecure: true,
+				},
+			},
+		},
+		{
+			name: "InsecureSkipVerify is false",
+			args: args{
+				insecure: false,
+			},
+			want: want{
+				obj: &T{
+					insecure: false,
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
should we run test in parallel?

Reply:
what do you mean??

Reply:
```suggestion
	for _, tc := range tests {
		test := tc
		t.Run(tt, func() {
			tt.Parallel()
		})
```
do we have to like above?

Reply:
understood thanks!